minimesos/src/test/java/com/containersol/minimesos/HelloWorldContainer.java

```diff
@@ -28,8 +29,15 @@ protected void pullImage() {
 
     @Override
     protected CreateContainerCmd dockerCommand() {
-        return dockerClient.createContainerCmd(HELLO_WORLD_IMAGE).withPrivileged(true)
-                .withName( getName() );
+        ExposedPort exposedPort = ExposedPort.tcp(SERVICE_PORT);
+        Ports portBindings = new Ports();
+        portBindings.bind(exposedPort, Ports.Binding(SERVICE_PORT));
+        return dockerClient.createContainerCmd(HELLO_WORLD_IMAGE)
+                .withEnv(String.format("SERVICE_%d_NAME=%s", SERVICE_PORT, SERVICE_NAME))
+                .withPrivileged(true)
+                .withName(getName())
+                .withPortBindings(portBindings)
+                .withExposedPorts(exposedPort);
     }
 
```

Comment:
Starting containers with the same name is the entire point. Consul will then return multiple A records, in random order! Awesome.

This is not supported yet and we should talk about it next week. Currently, Consul binds to the docker0 IP on port 53. We need some kind of way to discover the service discovery system... This solution does not support multiple instances of Consul though. Let's talk about alternatives Monday. For now, this is OK.